mix.exs

```diff
@@ -30,13 +31,34 @@ defmodule Guardian.DB.Mixfile do
   defp elixirc_paths(_), do: ["lib"]
 
   defp deps do
-    [
+    deps = [
       {:guardian, "~> 1.0"},
-      {:ecto, "~> 2.2"},
-      {:postgrex, "~> 0.13", optional: true},
       {:ex_doc, ">= 0.0.0", only: :dev, runtime: false},
-      {:credo, ">= 0.0.0", only: [:dev, :test], runtime: false}
+      {:credo, ">= 0.0.0", only: [:dev, :test], runtime: false},
+      {:jason, "~> 1.1", optional: true}
     ]
+
+    if System.get_env("ECTO3") do
```

Comment:
@ueberauth/core I leave this to you.

I am not a fan of doing such of practices where package authors maintain multiple versions of the dependencies.

Maintaining this becomes an issue,

1. Issues related to old packages become**our issues** and we could potentially keep growing the source code because of this, today is this simple change, tomorrow is a 1K line of code as I saw in other ecosystems.
2. Create the habit of being outdated on packages, relying on **our responsibilities** to keep backward compatibility even when the underline dependencies are not.

I truly believe that they have to decide to either keep the packages lockdown to a specific version or assume the responsibilities of breaking changes if they want to upgrade.

I wouldn't like to support `ecto <3.0` related issues and add complexity to the code because people would like to upgrade Guardian but they don't want to update Ecto.

I totally understand your pains if you are in legacy codes, but at the same time I believe that habits are the issue.

Reply:
If Ecto 3 is not out, then we should wait until it is actually out then.

If for any reason they decide to go back to the old API so they do not break backward compatibility then we will need to update the source code again.

I would wait until it is actually released.

Reply:
Again, let's leave that up to @ueberauth/core. Our project is already using `{:ecto, "~> 3.0.0-rc.1"}` so maybe it is reasonable to have a release candidate of guardian_db that supports it?

Either way, we're using this branch directly until things settle down. It's enough to get it working.

Reply:
> However, I won't necessarily have time to adjust this pull request to whatever @ueberauth/core decides

Totally understandable and I am really grateful for your contribution, we will probably work on top of your changes if we have to.

Thanks a lot for let us know.

Reply:
this isn't fun.. I think we need to add a clause in here which is something  like

`{:ecto, ">= 2.2.6 or ~> 3.0"}`

Testing is going to be fun...

Reply:
We could put that `or` clause on line 50, but right now I just have `>=` instead of "~>" which should take care of that case.

I'm quite happy with the way this all works right now, so if you want it changed, I need some clear instruction -- or feel free to grab the PR [with hub apply mail](https://www.spreedly.com/2014/06/24/merge-pull-request-considered-harmful/) and revise yourself.

Reply:
> I'm quite happy with the way this all works right now, so if you want it changed, I need some clear instruction -- or feel free to grab the PR with hub apply mail and revise yourself.

@nathany it sounds like you should stick with your fork for now.  As non-paid open source maintainers we have to make decisions that take the entire community into consideration.  

It's unreasonable for us to merge in changes in which contributors aren't open to incorporating feedback or helping with maintenance.

@ueberauth/core I suggest we close this PR

Reply:
@doomspork I am open to incorporating feedback! At least today.

Looking at your project, I see open pull requests from some time ago and am just giving you a heads up that if you are going to take your sweet time deciding what to do, I am not necessarily going to be able to drop what I'm doing to come update this pull request. 

It seems that the people here don't actually have a clear direction on what should be done here to support multiple versions of ecto. I made a best effort thanks to help from @ericmj on Slack. If that isn't good enough for you then please tell me what is. Saying "this isn't good [or fun]" in itself is not helpful.

I keep linking to the same article to suggest that you absolutely **can** take this code and do what you want with it. I'm trying to help you become better at maintaining your open source project, but instead you want to just close my pull request? Great way to build a better community.

Reply:
@nathany I apologize to you and others that the _unpaid maintainers of this free software_ have been busy with jobs and taking care of their families.  

There's 3 people maintaining this software and unfortunately the universe saw fit to throw a wrench into all of our personal lives.  I apologize that my struggling to deal with a family member's terminal illness has resulted in PRs not merging as fast as you might like.

That all said, you've twice relied to the feedback above with a comment suggesting you don't have time to incorporate feedback and we might as well just do it ourselves.  This is troubling to me.

Reply:
> I see open pull requests from some time ago

I decided to take a look at these PRs you referenced.  We've responded to damn near every PR with requests for changes.  In most cases we've even bumped the PR asking for an update.  We _will never_ merge unfinished PRs and assume the risk and responsibility of fixing it.